src/TimeoutMigrationTool/Asp/AspTimeoutsSource.cs

```diff
@@ -0,0 +1,608 @@
+            var upperLimitForCuttOffToFilterGuidEntriesOut = maxCutoffTime.AddYears(100).ToString(partitionKeyScope);
+
+            var query = new TableQuery<TimeoutDataEntity>()
+                .Where(CombineFilters(
+                    CombineFilters(GenerateFilterCondition("PartitionKey", QueryComparisons.GreaterThanOrEqual, cuttOffTimeAsPartitionKeyScope),
+                        TableOperators.And,
+                        GenerateFilterCondition("PartitionKey", QueryComparisons.LessThanOrEqual, upperLimitForCuttOffToFilterGuidEntriesOut)
+                            ),
+                    TableOperators.And,
+                    GenerateFilterCondition("OwningTimeoutManager", QueryComparisons.Equal, endpointName)));
+
+            TableContinuationToken token = null;
+            CancellationToken queryCancellationToken = CancellationToken.None;
+            var totalNumberOfItems = 0;
+            int previousBatch = 1, currentBatch = 1, numberOfItemsInBatch = 0;
+            var batchNumberAndSizes = new List<(int batchNumber, int batchSize)>();
+
+            do
+            {
+                // will issue a table scan for now
+                var seg = await endpointTimeoutTable.ExecuteQuerySegmentedAsync(
+                        query: query,
+                        token: token,
+                        requestOptions: null,
+                        operationContext: null,
+                        cancellationToken: queryCancellationToken)
+                    .ConfigureAwait(false);
+                token = seg.ContinuationToken;
+
+                // with batching we can at least efficiently insert
+                var hideByPartitionBatch = new TableBatchOperation();
+                foreach (var entitiesInTheSamePartition in seg.Results.GroupBy(x => x.PartitionKey))
+                {
+                    foreach (var timeoutDataEntity in entitiesInTheSamePartition)
+                    {
+                        // entries with Guid as partition key should never be modified. We are only interested in the query entities
+                        if (Guid.TryParse(timeoutDataEntity.PartitionKey.AsSpan(), out _))
+                        {
+                            continue;
+                        }
+
+                        if (hideByPartitionBatch.Count == 100)
+                        {
+                            await endpointTimeoutTable.ExecuteBatchAsync(hideByPartitionBatch);
+                            hideByPartitionBatch.Clear();
+                        }
+
+                        // we don't want to preserve the etag and we always want to win (is that true?)
+                        timeoutDataEntity.ETag = "*";
+                        // Fix the hiding part
+                        timeoutDataEntity.OwningTimeoutManager = $"{AspConstants.MigrationOngoingPrefix}{timeoutDataEntity.OwningTimeoutManager}";
+
+                        hideByPartitionBatch.Add(TableOperation.Merge(timeoutDataEntity));
```

Comment:
Shouldn't we also check if the batch operation hit the 4MB limit while staying under the 100 items limit?

Reply:
We did some quick math and came to the conclusion with 100 entries mostly being a few datetime, guids and endpoint name plus the state being externalized to blob storage we don't think we will hit this limit in any way. Do you suggest we play extra safe? This would vastly complicate this code

Reply:
If the state is externalized then it's fine. I wasn't aware of that.

Reply:
We added some size checking stuff. Have a look